pkgs/applications/audio/g4music/default.nix

```
@@ -0,0 +1,53 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
+, desktop-file-utils
+, gst_all_1
+, gtk4
+, libadwaita
+, meson
+, ninja
+, pkg-config
+, vala
+, wrapGAppsHook
+}:
+stdenv.mkDerivation rec {
+  pname = "g4music";
+  version = "2.4";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "neithern";
+    repo = "g4music";
+    rev = "v${version}";
+    sha256 = "sha256-y3G34NQEKPc145jLr98I6GaflzzgMVCCFLdcJDVkn8I=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    desktop-file-utils
+    meson
+    ninja
+    pkg-config
+    vala
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    gtk4
+    libadwaita
+  ] ++ (with gst_all_1; [
+    gst-plugins-base
+    gst-plugins-good
+    gstreamer
+  ]);
+
+  meta = with lib; {
+    description = "A beautiful, fast, fluent, light weight music player written in GTK4";
+    homepage = "https://gitlab.gnome.org/neithern/g4music";
+    license = licenses.gpl3;
```

```suggestion
    license = licenses.gpl3Only;
```

going by <https://gitlab.gnome.org/neithern/g4music/-/blob/master/COPYING> and meson.build, unless you saw the author say "gpl3 _or later_" somewhere in which case `gpl3Plus`.